src/main/java/com/gildedrose/AgeableRepository.java

```diff
@@ -0,0 +1,82 @@
+package com.gildedrose;
+
+import com.gildedrose.goblins_grotto.Item;
+import com.gildedrose.items.*;
+
+import java.util.*;
+//separate file which is the hashmap construction, this would be imported similar to any database connection
+
+public class AgeableRepository implements DataRepository {
+
+    public Ageable get(String id) {
+        return ItemToAgeableMapper.get(id);
+    }
+
+    public Collection<Ageable> getAll() {
+        return ItemToAgeableMapper.getAll();
```

Comment:
When I think of Repository and Mapper objects, I think of the following two responsibilities:

* Repository: knows how to find and return Items.
* Mapper: knows how to map between the types in database columns (i.e. `INTEGER`, `STRING`, `DATETIME`, etc.) and Java types (i.e. `int`, `String`, `java.util.Date`, etc.).  This type of object is also common in serialization libraries for JSON and XML. See [Jackson](https://www.baeldung.com/jackson-object-mapper-tutorial) for an example.

This experience is based on using Java ORM's like [Spring Data](https://docs.spring.io/spring-data/jpa/docs/2.5.1/reference/html/#preface).

Suggestion: What if this repository were named `HashMapItemRepository` and took a HashMap of Items in its constructor?  Then it could be swapped out later with a `DatabaseItemRepository` class that took a database connection object in it's constructor.

Reply:
Oh. Is it called an `AgeableRepository` because that is the return type of it's finder methods? Because otherwise there is no `Item` type we can return otherwise (since we're not using inheritance)?

This is me catching up to what's going on.  Let me know if I'm getting it.

Reply:
That was the case, but we determined the name AgeableRepository was a bit confusing, so `ItemsRepository` was decided on instead. We do indeed have the repository take in a hashmap of items (ageables), shown in [this commit here](https://github.com/jramnani/GildedRoseJavaLLC/pull/24/commits/7488aa40e935b1f84179a6908bc1e47f49ef7d3d)!